src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -1856,21 +1856,31 @@ private void HandleSupervisorStop(SupervisorStopDirectivePassivation msg)
             // We only have to do this if we have R-E enabled
             if (!_rememberEntities)
                 return;
-            
+
             var id = _entities.EntityId(msg.Child);
             // Just return if the child actor is not a recorded shard entity
-            if (id is null) 
+            if (id is null)
                 return;
-            
-            // Remove the child actor from the entity list
-            _entities.RemoveEntity(id);
```

Comment:
This is the bug - we don't want to remove the actor just yet. We want to mark it as passivating and let the normal termination / death watch process do that. That way we can't accidentally try to recreate a terminating actor if there are messages pending for it.